### PR TITLE
Pre-populate dialog with default location and name in open-in-notebook

### PR DESCRIPTION
**Pull request by jimmc**

### CODE REVIEW

File: sources/web/datalab/polymer/components/directory-picker-dialog/directory-picker-dialog.ts [hunk 1]
```diff
@@ -81,6 +92,21 @@ class DirectoryPickerDialogElement extends BaseDialogElement {
     return this._memoizedTemplate;
   }
 
+  ready() {
+    if (!this.readyPromise) {
+      super.ready();
+      this.readyPromise = this.$.filePicker.ready();
+      this.addEventListener('iron-overlay-opened',
```
Comment (yebrahim) [inline sources/web/datalab/polymer/components/directory-picker-dialog/directory-picker-dialog.ts]: A comment will be very useful here about why we're doing this.

File: sources/web/datalab/polymer/components/directory-picker-dialog/directory-picker-dialog.ts [hunk 2]
```diff
@@ -81,6 +92,21 @@ class DirectoryPickerDialogElement extends BaseDialogElement {
...
+        () => { this.resizeHandler(); });
+    }
+    return this.readyPromise;
+  }
+
+  resizeHandler() {
```
Comment (yebrahim) [inline sources/web/datalab/polymer/components/directory-picker-dialog/directory-picker-dialog.ts]: Do we need this handler, or can we just call the resizeHandler of the picker in the event listener above?

File: sources/web/datalab/polymer/modules/template-manager/template-manager.ts [hunk 3]
```diff
@@ -92,15 +92,43 @@ class TemplateManager {
 
   public static async newNotebookFromTemplate(template: NotebookTemplate) {
 
+    const appSettings = await SettingsManager.getAppSettingsAsync();
+
+    // TODO(jimmc): Look for a user preference for tempRoot
+    let tempRoot;
+    if (appSettings.defaultFileManager) {
```
Comment (yebrahim) [inline sources/web/datalab/polymer/modules/template-manager/template-manager.ts]: These 7 lines can be shortened: `let tempRoot = appSettings.DefaultFileManager || 'drive' + ':'`.

File: sources/web/datalab/polymer/modules/template-manager/template-manager.ts [hunk 4]
```diff
@@ -92,15 +92,43 @@ class TemplateManager {
 
   public static async newNotebookFromTemplate(template: NotebookTemplate) {
 
+    const appSettings = await SettingsManager.getAppSettingsAsync();
+
+    // TODO(jimmc): Look for a user preference for tempRoot
+    let tempRoot;
```
Comment (yebrahim) [inline sources/web/datalab/polymer/modules/template-manager/template-manager.ts]: Why is this "temp"?

File: sources/web/datalab/polymer/modules/utils/utils.ts [hunk 5]
```diff
@@ -190,6 +197,12 @@ class Utils {
     });
     return flatFields;
   }
+
+  public static async sleep(millis: number) {
```
Comment (yebrahim) [inline sources/web/datalab/polymer/modules/utils/utils.ts]: Are we using this somewhere? It looks like it's a debug utility?

File: sources/web/datalab/polymer/components/directory-picker-dialog/directory-picker-dialog.ts (hunk 1 above)
Comment (jimmc) [reply sources/web/datalab/polymer/components/directory-picker-dialog/directory-picker-dialog.ts]: Done.

File: sources/web/datalab/polymer/components/directory-picker-dialog/directory-picker-dialog.ts (hunk 2 above)
Comment (jimmc) [reply sources/web/datalab/polymer/components/directory-picker-dialog/directory-picker-dialog.ts]: I think having a resize handler here makes it more clear as to what the intent is, and is less likely to get overlooked if someone needs to do something else on a resize event.

File: sources/web/datalab/polymer/modules/template-manager/template-manager.ts (hunk 4 above)
Comment (jimmc) [reply sources/web/datalab/polymer/modules/template-manager/template-manager.ts]: It represents the root of where the temp file will go. I changed it to baseDir as being perhaps more clear.

File: sources/web/datalab/polymer/modules/template-manager/template-manager.ts (hunk 3 above)
Comment (jimmc) [reply sources/web/datalab/polymer/modules/template-manager/template-manager.ts]: Done.

File: sources/web/datalab/polymer/modules/template-manager/template-manager.ts [hunk 6]
```diff
@@ -92,15 +92,37 @@ class TemplateManager {
 
   public static async newNotebookFromTemplate(template: NotebookTemplate) {
 
+    const appSettings = await SettingsManager.getAppSettingsAsync();
+
+    // TODO(jimmc): Look for a user preference for tempRoot
```
Comment (yebrahim) [inline sources/web/datalab/polymer/modules/template-manager/template-manager.ts]: Update comment about `tempRoot`.

Comment (jimmc) [reply sources/web/datalab/polymer/modules/template-manager/template-manager.ts]: Done.

**Pull request merged**